services/mounts/service.go

```diff
@@ -0,0 +1,141 @@
+/*
+   Copyright The containerd Authors.
+
+   Licensed under the Apache License, Version 2.0 (the "License");
+   you may not use this file except in compliance with the License.
+   You may obtain a copy of the License at
+
+       http://www.apache.org/licenses/LICENSE-2.0
+
+   Unless required by applicable law or agreed to in writing, software
+   distributed under the License is distributed on an "AS IS" BASIS,
+   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+   See the License for the specific language governing permissions and
+   limitations under the License.
+*/
+
+package mounts
+
+import (
+	"context"
+	"errors"
+	"fmt"
+
+	mountsapi "github.com/containerd/containerd/api/services/mounts/v1"
+	"github.com/containerd/containerd/errdefs"
+	"github.com/containerd/containerd/log"
+	"github.com/containerd/containerd/mountmanager"
+	"github.com/containerd/containerd/plugin"
+	"github.com/containerd/containerd/protobuf"
+	"github.com/containerd/containerd/services"
+	"github.com/containerd/ttrpc"
+	"github.com/containerd/typeurl/v2"
+	"google.golang.org/grpc"
+)
+
+func init() {
+	plugin.Register(&plugin.Registration{
+		Type: plugin.GRPCPlugin,
+		ID:   services.MountsService,
```

Comment:
A pattern used elsewhere is to have one `ServicePlugin` which actually offers the functionality, and then another `GRPCPlugin` which exposes it via GRPC. This allows internal containerd components to have access to the service without going over GRPC. We should probably do the same thing here.

For example you can look at the tasks plugin:
- https://github.com/ambarve/containerd/blob/6c8aebe30cbd39a70f0b620af21b2ad91a4c62ae/services/tasks/local.go#L77
- https://github.com/ambarve/containerd/blob/6c8aebe30cbd39a70f0b620af21b2ad91a4c62ae/services/tasks/service.go#L35